modules/internal/ChapelDistribution.chpl

```diff
@@ -90,6 +90,10 @@ module ChapelDistribution {
       return (count==0);
     }
 
+    //
+    // TODO: Are there opportunities to optimize out the locking here,
+    // as on the add_arr() case?
```

Comment:
Note that array rank change slicing also requires a domain rank change slice, so this optimization would help any rank change slice, not just those on arrays.  Arguably rank-change slicing is less common than normal slicing, though (and more expensive in the distributed setting), so the "less common" comment may still apply.  I've added a modified comment to that effect.